src/library/Connect/KubernetesRemoteEnvironmentManager.cs

```diff
@@ -1197,6 +1229,27 @@ private async Task<bool> _WaitForDeploymentChangedAsync(string namespaceName, st
                 dirty = true;
             }
 
+            if (localProcessConfig?.IsProbesDisabled == true)
+            {
+                if (statefulSet.Spec.Template.Spec.Containers[containerIndex].LivenessProbe != null)
+                {
+                    patch.Remove(d => d.Spec.Template.Spec.Containers[containerIndex].LivenessProbe);
+                    reversePatch.Add(d => d.Spec.Template.Spec.Containers[containerIndex].LivenessProbe, statefulSet.Spec.Template.Spec.Containers[containerIndex].LivenessProbe);
+                }
+
+                if (statefulSet.Spec.Template.Spec.Containers[containerIndex].ReadinessProbe != null)
+                {
+                    patch.Remove(d => d.Spec.Template.Spec.Containers[containerIndex].ReadinessProbe);
+                    reversePatch.Add(d => d.Spec.Template.Spec.Containers[containerIndex].ReadinessProbe, statefulSet.Spec.Template.Spec.Containers[containerIndex].ReadinessProbe);
+                }

```

Comment:
Believe it or not, this function already has this logic around line 1172 :) I had done some research around this for this issue https://github.com/Azure/Bridge-To-Kubernetes/issues/114 and pasted some of the dev related findings on it. For some reason this code path which tights to _PatchStatefulSetAsync has probes removed but the other code paths dont. I will leave more details in the overall review comment, but for this part I would suggest removing old code that removes liviness and readiness probe in this function and keep this new block which incluses startupprobe as well.

Reply:
Done :)

How about this code part, do you know why StartupProbes are not set to null here?
https://github.com/Azure/Bridge-To-Kubernetes/blob/59c7d82d03668b322d9270f497e31ac7b6521333/src/library/Connect/KubernetesRemoteEnvironmentManager.cs#L486

Reply:
no, unfortunately I dont have the historical reasons on this part. However, I think it should be save to delete startup probes as well.

Reply:
Ok, I have removed the startup probe aswell